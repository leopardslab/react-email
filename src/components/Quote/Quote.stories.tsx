import { ComponentStory, ComponentMeta } from '@storybook/react';

import { Quote } from './Quote';

export default {
  component: Quote,
} as ComponentMeta<typeof Quote>;

//“template” of how args map to rendering
const Template: ComponentStory<typeof Quote> = (args) => <Quote {...args} />;

export const Default = Template.bind({});

Default.args = {
  children: 'This is a Text Quote',
};
