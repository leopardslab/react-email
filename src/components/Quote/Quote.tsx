import { ReactNode } from 'react';
import { makeStyles } from '../../utils/makeStyles';
import { BaseStyleProp } from '../types';

type QuoteStyles = 'root';

export interface QuoteProps extends BaseStyleProp<QuoteStyles> {
  children?: ReactNode;
}

const useStyles = makeStyles({
  root: { margin: 0, padding: 0 },
});

export const Quote = ({ children, classes, className }: QuoteProps) => {
  const styles = useStyles({ classes });

  return (
    <div style={styles.root} className={className}>
      {'>'} {children}
    </div>
  );
};
