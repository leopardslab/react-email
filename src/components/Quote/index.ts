export { Quote, QuoteProps } from './Quote';
