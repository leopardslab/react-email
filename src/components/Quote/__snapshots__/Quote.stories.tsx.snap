// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Quote Default smoke-test 1`] = `
<div style="margin: 0px auto; max-width: 600px;">
  <div id="quote"
       style="margin: 0px; padding: 6px 6px 6px 10px; border-left: 4px solid rgb(208, 215, 222); color: rgb(126, 134, 152); font-style: italic; font-size: 1.1em;"
  >
    This is a Text Quote Component written in TypeScript
  </div>
</div>
`;
